### ISSUE TITLE
`docker build` with several `--tag` options depends on tags order

### ISSUE THREAD

**Issue by 907th** [docker/docker#19297]
There is a difference between
```
$ docker build --tag name --tag registry:5000/name .
# <- only 'registry:5000/name' tag created, 'name' tag NOT created
```
and
```
docker build --tag registry:5000/name --tag name .
# <- both 'registry:5000/name' and 'name' tags created
```

**Comment by cpuguy83** [comment]
This is because `-t` does not accept multiple entries.
It's really only ever tagging 1 image both times (should be the last one specified).
In your test you've already created `registry:5000/name` in the first run, and the 2nd run tags `name`

Note that docker 1.10 will accept multiple entires for `--tag`.

**Comment by 907th** [comment]
@cpuguy83 Great! Thank you. I thought it was released in 1.9 already.

**Comment by thaJeztah** [comment]
PR is here; https://github.com/docker/docker/pull/15780
